### PR TITLE
Fix issue #26568 about customer blank schema

### LINKED ISSUE #26568: webservice customers - can't create new customers
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md)
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add screenshots

Impossible to create new customer via webservices

URL/api/customers?schema=blank gives an error : 
```
<prestashop>
<errors>
<error>
<code>5</code>
<message>
[PHP Notice #8] Trying to access array offset on value of type null (/home/.../public_html/classes/webservice/WebserviceOutputBuilder.php, line 669)
</message>
</error>
<error>
<code>5</code>
<message>
[PHP Notice #8] Trying to access array offset on value of type null (/home/.../public_html/classes/webservice/WebserviceOutputBuilder.php, line 716)
</message>
</error>
</errors>
</prestashop>
```

### Expected behavior

It should done something like : 
```
<prestashop>
 <customer>
  <id/>
  <id_default_group/>
  <id_lang/>
  <newsletter_date_add/>
  <ip_registration_newsletter/>
  <last_passwd_gen/>
  <secure_key/>
  <deleted/>
  <passwd/>
  <lastname/>
  <firstname/>
  <email/>
  <id_gender/>
  <birthday/>
  <newsletter/>
  <optin/>
  <website/>
  <company/>
  <siret/>
  <ape/>
  <outstanding_allow_amount/>
  <show_public_prices/>
  <id_risk/>
  <max_payment_days/>
  <active/>
  <note/>
  <is_guest/>
  <id_shop/>
  <id_shop_group/>
  <date_add/>
  <date_upd/>
  <reset_password_token/>
  <reset_password_validity/>
  <associations>
   <groups>
    <group>
     <id></id>
    </group>
   </groups>
  </associations>
 </customer>
</prestashop>
```

### Steps to reproduce

Create a new customer with webservice
> Originally posted by @ceer (see the [comment](https://github.com/PrestaShop/PrestaShop/pull/29258#issuecomment-1223126475))

Here’s is the way to reproduce the bug
1. Use this docker-compose:

```yaml
version: "3"

services:
  prestashop:
    image: prestashop/prestashop:1.7.8.4-7.4-apache
    environment:
      - DB_SERVER=db
      - DB_NAME=prestashop
      - DB_USER=prestashop
      - DB_PASSWD=prestashop
      - PS_DEV_MODE=0
      - PS_DEMO_MODE=0
      - PS_INSTALL_AUTO=1
      - PS_DOMAIN=localhost:8080
      - PS_FOLDER_ADMIN=admin
      - ADMIN_MAIL=demo@prestashop.com
      - ADMIN_PASSWD=prestashop_demo
    links:
      - db
    depends_on:
      - db
    ports:
      - 8080:80
    networks:
      - prestashop-net

  db:
    image: mysql:5.7
    command: --default-authentication-plugin=mysql_native_password
    environment:
      - MYSQL_ROOT_PASSWORD=admin
      - MYSQL_DATABASE=prestashop
      - MYSQL_USER=prestashop
      - MYSQL_PASSWORD=prestashop
    networks:
      - prestashop-net

networks:
  prestashop-net:
```

2. Enable Webservice in back-office, create a key with all permissions for `customers` ressource
3. Go to http://localhost:8080/api/customers?schema=blank

Results :
```xml
<prestashop xmlns:xlink="http://www.w3.org/1999/xlink">
    <errors>
        <error>
            <code>
                <![CDATA[ 5 ]]>
            </code>
            <message>
                <![CDATA[
[PHP Notice #8] Trying to access array offset on value of type null (/var/www/html/classes/webservice/WebserviceOutputBuilder.php, line 669)
]]>
            </message>
        </error>
        <error>
            <code>
                <![CDATA[ 5 ]]>
            </code>
            <message>
                <![CDATA[
[PHP Notice #8] Trying to access array offset on value of type null (/var/www/html/classes/webservice/WebserviceOutputBuilder.php, line 716)
]]>
            </message>
        </error>
    </errors>
</prestashop>
```

### PrestaShop version(s) where the bug happened

1.7.8.0

### PHP version(s) where the bug happened

7.4

### If your bug is related to a module, specify its name and its version

_No response_

### PR DESCRIPTION
**Pull request by ceer**
Pull Request #28280 does not fix totally this PHP 7.4 bug described in issue #26568 

| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Fix PHP 7.4 Notice error `Trying to access array offset on value of type null (/home/.../public_html/classes/webservice/WebserviceOutputBuilder.php, line 669)` when requesting `/api/customers?schema=blank`
| Type?             | bug fix
| Category?         | WS
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | Fixes #26568
| Related PRs       | PR #28280
| How to test?      | test /api/customers?schema=blank with PHP 7.4
| Possible impacts? | none

### CODE REVIEW

**Review by matks — COMMENTED** [review]
Hello @ceer and thank you for the contribution. I think you found a good thing to fix 😉 but for me to verify it I need to 100% understand how to reproduce the bug.

In https://github.com/PrestaShop/PrestaShop/issues/26568 we did not successfully gathered "how to..." steps to reproduce the bug. If I cannot reproduce I cannot verify 😭 

Could you please open a new bug report with very detailed "how to..." steps 😉 something we can follow, reproduce the bug, then use your PR and verify that your PR solves the bug?

Thank you

**Comment by ceer** [comment]
Hi @matks 
Here’s is the way to reproduce the bug
1. Use this docker-compose:

```yaml
version: "3"

services:
  prestashop:
    image: prestashop/prestashop:1.7.8.4-7.4-apache
    environment:
      - DB_SERVER=db
      - DB_NAME=prestashop
      - DB_USER=prestashop
      - DB_PASSWD=prestashop
      - PS_DEV_MODE=0
      - PS_DEMO_MODE=0
      - PS_INSTALL_AUTO=1
      - PS_DOMAIN=localhost:8080
      - PS_FOLDER_ADMIN=admin
      - ADMIN_MAIL=demo@prestashop.com
      - ADMIN_PASSWD=prestashop_demo
    links:
      - db
    depends_on:
      - db
    ports:
      - 8080:80
    networks:
      - prestashop-net

  db:
    image: mysql:5.7
    command: --default-authentication-plugin=mysql_native_password
    environment:
      - MYSQL_ROOT_PASSWORD=admin
      - MYSQL_DATABASE=prestashop
      - MYSQL_USER=prestashop
      - MYSQL_PASSWORD=prestashop
    networks:
      - prestashop-net

networks:
  prestashop-net:
```

2. Enable Webservice in back-office, create a key with all permissions for `customers` ressource
3. Go to http://localhost:8080/api/customers?schema=blank

Results :
```xml
<prestashop xmlns:xlink="http://www.w3.org/1999/xlink">
    <errors>
        <error>
            <code>
                <![CDATA[ 5 ]]>
            </code>
            <message>
                <![CDATA[
[PHP Notice #8] Trying to access array offset on value of type null (/var/www/html/classes/webservice/WebserviceOutputBuilder.php, line 669)
]]>
            </message>
        </error>
        <error>
            <code>
                <![CDATA[ 5 ]]>
            </code>
            <message>
                <![CDATA[
[PHP Notice #8] Trying to access array offset on value of type null (/var/www/html/classes/webservice/WebserviceOutputBuilder.php, line 716)
]]>
            </message>
        </error>
    </errors>
</prestashop>
```

**Review by kpodemski — DISMISSED** [review]

File: classes/webservice/WebserviceOutputBuilder.php [hunk 1]
```diff
@@ -666,7 +666,7 @@ protected function renderAssociations($object, $depth, $associations, $ws_params
                             $value = $object_assoc;
                         }
                         if (empty($fields_assoc)) {
-                            $fields_assoc = [['id' => $value['id']]];
+                            $fields_assoc = [['id' => (isset($value['id']) ? $value['id'] : null)]];
```
Comment (atomiix) [inline classes/webservice/WebserviceOutputBuilder.php]:
```suggestion
                            $fields_assoc = [['id' => ($value['id'] ?? null)]];
```

**Review by atomiix — DISMISSED** [review]
Just a small suggestion but no mandatory, still approving ;)

**Comment by MatShir** [comment]
Hi, It seems that your fix is applied for the develop branch (so for the next 8.1). Since the issue is a regression, you can switch branch to 1.7.8.x, I would be nice to have your fix in the next patch release :)

**Comment by ceer** [comment]
Yes @MatShir it will be great. If so, this one regarding php 7.4 also: https://github.com/PrestaShop/PrestaShop/pull/28280

**Comment by MatShir** [comment]
Once your PR target the right branch (1.7.8.x instead of develop), I will push for some reviews :) 
Capture d’écran 2022-09-27 à 10 58 20

**Comment by kpodemski** [comment]
Hi @ceer !

If I recall correctly, as there's 8.0 released, you could rebase into 8.0.x branch to include this fix inside 8.0.1

but let's wait for @MatShir confirmation

**Comment by kpodemski** [comment]
Hi @ceer 

I can see that you rebased your PR and then we didn't have a chance to review it. Could you rebase it **and** ping me so I can help you move things forward so it could go to the core?

**Comment by mflasquin** [comment]
New PR targeting 8.0.x branch : https://github.com/PrestaShop/PrestaShop/pull/30993/files

**Pull request closed without merging**